text/3606-temporary-lifetimes-in-tail-expressions.md

```diff
@@ -0,0 +1,163 @@
+# Temporary lifetimes in tail expressions
+
+- Feature Name: `tail_temporaries_shorter_lifetime`
+- Start Date: 2023-05-04
+- RFC PR: [rust-lang/rfcs#3606](https://github.com/rust-lang/rfcs/pull/3606)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+
+In the next edition, drop temporaries in tail expressions *before* dropping locals, rather than after.
+
+![before-after](https://hackmd.io/_uploads/HyVB0FtkA.svg)
```

Comment:
This diagram is helpful, and while it's technically redundant, I think that it should be properly described either in the alt text (difficult, because markdown isn't designed for this) or with a caption below describing it. Would be nice to screen reader users & folks reading the text-only version both.

Reply:
Oh, forgot the alt text! Oops. Fixed!